src/components/common/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,52 @@
+import { MdSearch } from 'react-icons/md';
+
+import IconWrapper from '../IconWrapper';
+import Input from '../Input';
+import { SearchBarWrapper } from './SearchBar.style';
+import { SearchBarProps } from './SearchBar.type';
+
+/**
+ *
+ * @description SearchBar 컴포넌트
+ * @summary 사용법 :
+      <SearchBar
+        handleSearchIcon={() => alert('검색됨!')}
+        handleFormSubmit={handleSubmit(onSubmit)}
+        width={'500px'}
+      />
+
+ * @param handleSearchIcon : 선택 | 함수. 검색 아이콘에 들어갈 클릭 이벤트를 받음.
+ * @param handleFormSubmit : 선택 | 함수. input에 검색어를 입력 후 엔터 할 때 submit 이벤트를 받음 (react-hook-form의 handleSubmit 함수와 연관됨.)
+ * @param maxWidth : 선택 | 문자열 타입. 검색바 컴포넌트의 최대 너비를 의미함. (px, rem, %를 다 받습니다.)
+ * @param …props : 커스텀을 위함 (+ react-hook-form)
+ * @returns
+ */
+
+const SearchBar = ({
+  handleSearchIcon,
+  handleFormSubmit,
+  maxWidth,
+  ...props
+}: SearchBarProps) => {
+  return (
+    <SearchBarWrapper $maxWidth={maxWidth}>
+      <form onSubmit={handleFormSubmit}>
```

Comment:
P3; form형식이라 엔터를 치면 새로고침이 되는 현상이 있더라구요! 
이거 따로 함수를 빼서 `e.preventDefault();` 코드를 추가해줘서 새로고침을 방지해주면 어떨까요? ㅎㅎ
```suggestion
const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (handleFormSubmit) {
      handleFormSubmit();
    }
  };

  return (
    <SearchBarWrapper $maxWidth={maxWidth}>
      <form onSubmit={handleSubmit}>
```

Reply:
좋은 것 같아요 !! 수정하겠습니다 :)